lib/config/defaults.js

```diff
@@ -709,7 +745,11 @@ const applyOutputDefaults = (
 	F(output, "path", () => path.join(process.cwd(), "dist"));
 	F(output, "pathinfo", () => development);
 	D(output, "sourceMapFilename", "[file].map[query]");
-	D(output, "hotUpdateChunkFilename", "[id].[fullhash].hot-update.js");
+	D(
+		output,
+		"hotUpdateChunkFilename",
+		`[id].[fullhash].hot-update.${output.module ? "mjs" : "js"}`
```

Comment:
Maybe we can read `package.json` here and use `js` when `"type": "module"` (same for `filename`)?